### PR TITLE
Add initial business logic for GetMatch endpoint

### PR DESCRIPTION
**Pull request by echappen**
## What’s changing?

- Adds initial business logic for `GetMatch` endpoint
- Adds unit and integration tests for the Match Resolution API function app
- Adds tests to CI code coverage

## Why?

Relates to Jira Ticket NAC-165

This implements the [Get Match Details endpoint](https://github.com/18F/piipan/blob/dev/match/docs/openapi/resolution/match_show.yaml) from the Match Resolution API schema.
Authorization for this endpoint hasn't been done yet—I wanted to confirm which authorizations would be needed. Should we allow states not affiliated with the match to view a match's details?

Tagging @bgfoshay as primary reviewer, but as always please tag other Ventera Engs as needed.
Tagging @kingcomma to confirm what authorizations should be implemented.

## This PR has:

- [x] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)
- [x] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)
- [x] **Automated unit tests** (_to maintain or increase level of code coverage_)
- [ ] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

## Anything else?

To avoid code duplication, the API integration tests borrow the `DbFixture` from `Piipan.Match.Core.IntegrationTests`. At some point, we should explore how to make it easier to share supporting functionality between test suites.

### CODE REVIEW

**Comment by echappen** [comment]
Tests are passing but not sure why Code Coverage is failing.

File: match/src/Piipan.Match/Piipan.Match.Func.ResolutionApi/GetMatchApi.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            _matchResAggregator = matchResAggregator;
+        }
+
+        [FunctionName("GetMatch")]
+        public async Task<IActionResult> GetMatch(
+            [HttpTrigger(AuthorizationLevel.Function, "get", Route = "matches/{matchId}")] HttpRequest req,
+            string matchId,
+            ILogger logger)
+        {
+            LogRequest(logger, req);
```
Comment (bgfoshay) [inline match/src/Piipan.Match/Piipan.Match.Func.ResolutionApi/GetMatchApi.cs:46]: Thought: Is this something that should this be handled by our middleware?

File: match/src/Piipan.Match/Piipan.Match.Func.ResolutionApi/Models/ApiErrorResponse.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace Piipan.Match.Func.ResolutionApi.Models
+{
+    /// <summary>
+    /// Represents a generic error response for an API request
+    /// </summary>
+    public class ApiErrorResponse
```
Comment (bgfoshay) [inline match/src/Piipan.Match/Piipan.Match.Func.ResolutionApi/Models/ApiErrorResponse.cs:9]: Question: Should we consider defining this class in Shared library/project? It's generic enough that I can see it being reused in the future as a response for other APIs

File: match/src/Piipan.Match/Piipan.Match.Func.ResolutionApi/GetMatchApi.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            try
+            {
+                var match = _matchRecordDao.GetRecordByMatchId(matchId);
+                var matchResEvents = _matchResEventDao.GetEvents(matchId);
+                await Task.WhenAll(match, matchResEvents);
+                var matchResRecord = _matchResAggregator.Build(match.Result, matchResEvents.Result);
+                var response = new ApiResponse() { Data = matchResRecord };
+                return new JsonResult(response) { StatusCode = StatusCodes.Status200OK };
+            }
+            catch (InvalidOperationException ex)
```
Comment (bgfoshay) [inline match/src/Piipan.Match/Piipan.Match.Func.ResolutionApi/GetMatchApi.cs:56]: Thought: Along the same lines as my logging comments, we could consider introducing API exception handling middleware (in another ticket) to that handle generic errors like NotFound & InternalServer exceptions.

**Review by bgfoshay — APPROVED** [review]
Looks good. The changes got me thinking about potential re-use though. See my comments on maybe introducing some of this logic as Azure Function middleware

Thought (non-blocking): I think there could be some future benefit to moving some of the request logging & exception handling logic into Azure Function middleware (https://github.com/Azure/azure-functions-dotnet-worker/tree/main/samples/CustomMiddleware).

**Comment by ryangeorgeharvey** [comment]
Left a comment on FNS Teams regarding code coverage. Sorry if it's no longer relevant.

File: match/src/Piipan.Match/Piipan.Match.Func.ResolutionApi/Models/ApiErrorResponse.cs (hunk 2 above)
Comment (echappen) [reply match/src/Piipan.Match/Piipan.Match.Func.ResolutionApi/Models/ApiErrorResponse.cs:9]: Yes I think we should! I started [going down that route](https://github.com/18F/piipan/pull/2825) and found it to be a good amount of work, and I wasn't certain on the approach, so I set it aside in favor of getting an endpoint up quickly. But I'd like to return to this.

File: match/tests/Piipan.Match.Func.ResolutionApi.IntegrationTests/GetMatchApiIntegrationTests.cs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+using Moq;
+using Npgsql;
+using Piipan.Match.Core.Builders;
+using Piipan.Match.Core.DataAccessObjects;
+using Piipan.Match.Core.Models;
+using Piipan.Shared.Database;
+using Piipan.Match.Core.IntegrationTests;
+using Piipan.Match.Func.ResolutionApi.Models;
+
+namespace Piipan.Match.Func.ResolutionApi.IntegrationTests
```
[the remark this answers is not in the document]
Comment (echappen) [reply match/tests/Piipan.Match.Func.ResolutionApi.IntegrationTests/GetMatchApiIntegrationTests.cs:18]: Yes, I think that's a good idea. Should I attempt in this PR or create a new issue?

[the remark this answers is not in the document]
Comment (bgfoshay) [reply match/tests/Piipan.Match.Func.ResolutionApi.IntegrationTests/GetMatchApiIntegrationTests.cs:18]: I think this could be a new issue.

**Pull request merged**